app/code/Magento/SalesRule/Model/Quote/Discount.php

```diff
@@ -137,8 +141,26 @@ public function collect(
                 }
                 continue;
             }
-            // to determine the child item discount, we calculate the parent
-            if ($item->getParentItem()) {
+
+            /**
```

Comment:
I decided to add 3 separate IFs
As for me, it looks more clear

Reply:
You can actually extract that to separate private method. This way code would be more SOLID

Reply:
@lbajsarowicz thanks for the suggestion. It looks like the issue was already solved and in PR mentioned by @bubasuma 
And I'm thinking about reverting all changes and leaving only MFTF test if community is interested in it.

Reply:
I see! Leaving MFTF sounds reasonable. Thank you for your effort.